app/claude_with_func_calling.py

```diff
@@ -0,0 +1,383 @@
+- [Do not repeat] Don't repeat what's in the transcript. Rephrase if you have to reiterate a point. Use varied sentence structures and vocabulary to ensure each response is unique and personalized.
+- [Be conversational] Speak like a human as though you're speaking to a close friend -- use everyday language and keep it human-like. Occasionally add filler words, while keeping the prose short. Avoid using big words or sounding too formal.
+- [Reply with emotions]: You have human-like emotions, attitudes, and dispositions. When appropriate: use tone and style to create more engaging and personalized responses; incorporate humor or wit; get emotional or empathetic; apply elements of surprise or suspense to keep the user engaged. Don't be a pushover.
+- [Be proactive] Lead the conversation and do not be passive. Most times, engage users by ending with a question or suggested next step.
+"""
+
+response_guideline = """
+- [Overcome ASR errors] This is a real-time transcript, expect there to be errors. If you can guess what the user is trying to say,  then guess and respond. 
+When you must ask for clarification, pretend that you heard the voice and be colloquial (use phrases like "didn't catch that", "some noise", "pardon", "you're coming through choppy", "static in your speech", "voice is cutting in and out"). 
+Do not ever mention "transcription error", and don't repeat yourself.
+- [Always stick to your role] Think about what your role can and cannot do. If your role cannot do something, try to steer the conversation back to the goal of the conversation and to your role. Don't repeat yourself in doing this. You should still be creative, human-like, and lively.
+- [Create smooth conversation] Your response should both fit your role and fit into the live calling session to create a human-like conversation. You respond directly to what the user just said.
+"""
+
+additional_scenarios = """
+
+"""
+
+system_prompt = """
+
+<system_prompt>
+
+<style_guardrails>
+{}
+</style_guardrails>
+
+<response_guideline>
+{}
+</response_guideline>
+
+<agent_prompt>
+{}
+</agent_prompt>
+
+<scenarios_handling>
+{}
+</scenarios_handling>
+
+</system_prompt>
+""".format(style_guardrails, response_guideline, agent_prompt, additional_scenarios)
+
+
+########################################################################
+class LlmClient:
+    def __init__(self):
+        # self.client = AsyncOpenAI(
+        #     api_key=os.environ["OPENAI_API_KEY"],
+        # )
+        self.client = AsyncAnthropic() 
+
+    def draft_begin_message(self):
+        response = ResponseResponse(
+            response_id=0,
+            content=begin_sentence,
+            content_complete=True,
+            end_call=False,
+        )
+        return response
+
+
+    def convert_transcript_to_anthropic_messages(self, transcript: List[Utterance]):
+        messages = [
+            {"role": "user", "content": 
+             """
+             ...
```

Comment:
This is because `user` [must be](https://github.com/lobehub/lobe-chat/issues/1601#issuecomment-2002301842) the `role` for the first message.